vignettes/caseStudy2-NutrientThreshold.Rmd

```diff
@@ -0,0 +1,443 @@
+            axis.title.y=element_blank(),
+            axis.text.y=element_blank(),
+            axis.ticks.y=element_blank(),
+        )
+    }
+    if (ncol(matrix.A) >= 20){
+        # too many resources
+        fig <- fig + theme(
+            axis.title.x=element_blank(),
+            axis.text.x=element_blank(),
+            axis.ticks.x=element_blank()
+        )
+    }
+    fig
+}
+makeUMAP <- function(matrix, n_neighbors=10, min_dist=0.1, gradient=NULL, gradient_title = 'gradient', group=NULL, group2=NULL){
+    custom.config = umap.defaults
+    custom.config$n_neighbors = n_neighbors
+    custom.config$min_dist = min_dist
+
+    df <- as.data.frame(umap(matrix,config = custom.config)$layout)
+    df$gradient <- gradient
+
+    if (is.null(gradient)){
+        df$gradient <- 1
+
+    }
+    colnames(df) = c('UMAP_2', 'UMAP_1', gradient_title)
+    if (is.null(group)){
+        ggplot(df, aes_string('UMAP_2', 'UMAP_1', color=gradient_title)) +
+            geom_point() +
+            scale_color_gradient(low="blue", high="red")
+    } else {
+        if (is.null(group2)){
+            ggplot(df, aes_string('UMAP_2', 'UMAP_1', color=gradient_title)) +
+                geom_point(aes(color = group)) + theme_bw()
+        } else {
+            ggplot(df, aes_string('UMAP_2', 'UMAP_1', color=gradient_title)) +
+                geom_point(aes(color = group, shape = group2)) + theme_bw()
+        }
+    }
+}
+
+makeHeatmap(as.matrix(dist.community.initial.df),
+            title = "dissimilarity matrix",
+            x.label = "community.1",
+            y.label = "community.2")
+makeUMAP(matrix = community.initial.df,
+         n_neighbors = 5,
+         group = factor(seq_len(n.community)),
+         gradient_title = "community")
+
+```
+
+## initialize shared parameters
+
+```{r}
+crm_params <- list(
+    n_species = n_species,
+    n_resources = n_resources,
+    x0 = NULL,
+    E = E,
+    resources = rep(1,n_resources),
+    monod_constant = monod_constant,
+    migration_p = 0,
+    stochastic = FALSE,
+    t_start = 0,
+    t_end = 50,
+    t_step = 1,
+    t_store = t_store,
+    growth_rates = growth_rates,
+    norm=FALSE)
+```
+
+## generate resource gradients
+
+```{r}
+resourceConcentration <- 10^seq(0,4,1) # 1 to 10000
+n.medium <- 5
+density.medium <- 0.8
+set.seed(42)
+resource.initial.df <- gradient.df.generator(n_row = n.medium, n_col = n_resources, density_row = density.medium, max_gradient = 0.7, error_interval = 0.1)
+```
+
+## test one run
+
+```{r}
+crmExample <- simulateConsumerResource(
+    n_species = n_species,
+    n_resources = n_resources,
+    E = E,
+    x0 = as.numeric(community.initial.df[1,]),
+    resources = as.numeric(resourceConcentration[3]*resource.initial.df[1,]),
+    growth_rates = growth_rates,
+    monod_constant = monod_constant,
+    stochastic = FALSE,
+    t_end = 50,
+    t_step = 1,
+    t_store = 50,
+    norm = FALSE)
+makePlot(crmExample$matrix)
+makePlotRes(crmExample$resources)
+```
```

Comment:
crmExample is TreeSE (or should be), hence we can use existing miaViz tools and not reinvent the wheel here